File: backend/api/network.py
"""网络信息端点 — 供前端获取服务器局域网地址"""
from fastapi import APIRouter

router = APIRouter()

# 由 run.py 启动时注入
LAN_IP = "127.0.0.1"
PORT = 8000
FRONTEND_PORT = 3000


@router.get("/api/network")
async def get_network_info():
    return {
        "lanIp": LAN_IP,
        "port": PORT,
        "frontendPort": FRONTEND_PORT,
    }

File: frontend/src/services/network.ts
export interface NetworkInfo {
  lanIp: string;
  port: number;
  frontendPort: number;
}

/** 获取服务器网络信息 — 用于显示局域网访问地址 */
export async function fetchNetworkInfo(): Promise<NetworkInfo> {
  const resp = await fetch('/api/network');
  if (!resp.ok) {
    throw new Error(`Network info request failed: ${resp.status}`);
  }
  return resp.json();
}
